hca/define_api.py

```diff
@@ -1,37 +1,172 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-import json, sys, os
-from io import open
+import json
+import os
+import sys
 import requests
-from .parser import get_parser
+
+from io import open
+
+from .parser import get_parser, ADDED_COMMANDS
+from .constants import Constants
 
 
 class API:
     """Class for interacting with the REST Api."""
 
-    def __init__(self, api_url, user):
+    def __init__(self, test=False):
         """Initialize the CLI API."""
-        self.spec = self.get_spec()
-        self.base_url = self.spec['host'] + self.spec['basePath']
-        self.parser = get_parser(self.spec)
+        spec = self.get_spec(test)
```

Comment:
since you're passing in `test` as a constructor argument, consider just storing it in `self.test`, and using that instead of passing it around.

same comment applies about the ambiguity of this name.